components/clarinet-files/src/project_manifest.rs

```diff
@@ -243,16 +249,19 @@ impl ProjectManifest {
                                 }
                                 _ => DEFAULT_EPOCH,
                             };
-                            config_contracts.insert(
-                                contract_name.to_string(),
-                                ClarityContract {
-                                    name: contract_name.clone(),
-                                    code_source,
-                                    deployer,
-                                    clarity_version,
-                                    epoch,
-                                },
-                            );
+                            let clarity_contract = ClarityContract {
+                                name: contract_name.clone(),
+                                code_source,
+                                deployer,
+                                clarity_version,
+                                epoch,
+                            };
+                            config_contracts
+                                .insert(contract_name.clone(), clarity_contract.clone());
+
+                            let mut contract_location = project_root_location.clone();
+                            contract_location.append_path(contract_path)?;
+                            contracts_settings.insert(contract_location, clarity_contract);
```

Comment:
`contract_settings` basically hold the same data as `config_contracts` but with the location as the key.